spec/controllers/aspects_controller_spec.rb

```diff
@@ -50,30 +31,32 @@
     context "with valid params" do
       it "creates an aspect" do
         expect(alice.aspects.count).to eq(2)
-        post :create, "aspect" => {"name" => "new aspect"}
+        post :create, aspect: {name: "new aspect"}
         expect(alice.reload.aspects.count).to eq(3)
       end
-      it "redirects to the aspect's contact page" do
-        post :create, "aspect" => {"name" => "new aspect"}
-        expect(response).to redirect_to(contacts_path(:a_id => Aspect.find_by_name("new aspect").id))
+
+      it "returns the created aspect as json" do
+        post :create, aspect: {name: "new aspect"}
+        expect(JSON.parse( response.body )["id"]).to eq Aspect.find_by_name("new aspect").id
+        expect(JSON.parse( response.body )["name"]).to eq "new aspect"
       end
 
       context "with person_id param" do
         it "creates a contact if one does not already exist" do
           expect {
-            post :create, :format => 'js', :aspect => {:name => "new", :person_id => eve.person.id}
+            post :create, format: 'js', person_id: eve.person.id, aspect: {name: "new"}
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.